internal/apis/config/webhook/types.go

```diff
@@ -26,12 +26,14 @@ type WebhookConfiguration struct {
 	metav1.TypeMeta
 
 	// securePort is the port number to listen on for secure TLS connections from the kube-apiserver.
+	// If 0, a random available port will be chosen.
 	// Defaults to 6443.
-	SecurePort *int
+	SecurePort int32
 
 	// healthzPort is the port number to listen on (using plaintext HTTP) for healthz connections.
+	// If 0, a random available port will be chosen.
 	// Defaults to 6080.
-	HealthzPort *int
+	HealthzPort int32
```

Comment:
It's interesting that in the Kubernetes config file, port 0 **sometimes disables** the service:

field | description
-- | --
port **int32** | port is the port for the Kubelet to serve on. The port number must be between 1 and 65535, inclusive. Default: 10250
readOnlyPort **int32** | readOnlyPort is the read-only port for the Kubelet to serve on with no authentication/authorization. The port number must be between 1 and 65535, inclusive. Setting this field to 0 disables the read-only service. Default: 0 (disabled)
healthzPort **int32** | healthzPort is the port of the localhost healthz endpoint (set to 0 to disable). A valid number is between 1 and 65535. Default: 10248

https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/#kubelet-config-k8s-io-v1beta1-KubeletConfiguration

Reply:
Thanks for the note, I checked and it looks that here it indeed just takes a random port:
https://github.com/cert-manager/cert-manager/blob/master/pkg/webhook/server/server.go#L125
https://github.com/cert-manager/cert-manager/blob/master/pkg/webhook/server/server.go#L190